internal/common/common.go

```diff
@@ -140,17 +140,21 @@ func WriteFileAtomic(filename string, perm os.FileMode, data []byte) error {
 }
 
 var (
+	Hostname hostNameT
+)
+
+type hostNameT struct {
 	once     sync.Once
 	hostname string
-)
+}
 
-func GetHostname() string {
-	once.Do(func() {
+func (h *hostNameT)Get() string {
+	h.once.Do(func() {
 		var err error
-		hostname, err = os.Hostname()
+		h.hostname, err = os.Hostname()
 		if err != nil {
-			hostname = ""
+			h.hostname = ""
 		}
 	})
-	return hostname
+	return h.hostname
```

Comment:
Sorry, I wasn't clear. I mean something like:

``` go
var (
	h struct {
		once     sync.Once
		hostname string
	}
)

func GetHostname() string {
	h.once.Do(func() {
		log.Printf("hostname get called")
		var err error
		h.hostname, err = os.Hostname()
		if err != nil {
			h.hostname = ""
		}
	})
	return h.hostname
}


```
where the struct is totally hidden and you just call `common.GetHostname()`.

Anyway I don't think that calling os.Hostname is a big overhead so we should also avoid the sync once.

Reply:
Maybe Get os.Hostname once in init section.
Like




`
func init(){
	setHostname()
}
var (
	hostname string
)

func GetHostname() string {
	return hostname
}

func setHostname() {
	var err error
	hostname, err = os.Hostname()
	if err != nil {
		hostname = ""
	}
}
`
And return global var hostname from func GetHostname.

Reply:
Seems good.

Reply:
Update to one clear commit